### PR TITLE
[spirv] Move Adreno matmul CodeGen config as SPIR-V default

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

**Comment by antiagainst** [comment]
https://gist.github.com/iree-github-actions-bot/6f331175c348551df7165bfcbd1a9555 has the full benchmark results sorted in decresing absolute percent change order. Showing the top ones here:

| Benchmark Name                                                                                                                                                                                                                                                                                                | Average Latency (ms)      | Median Latency (ms) | Latency Standard Deviation (ms) |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------- | ------------------- | ------------------------------- |
| [MobileSSD [fp32] (TFLite) full-inference with IREE-Vulkan @ SM-G980F (GPU-Mali-G77)](https://perf.iree.dev/serie?IREE?MobileSSD%20[fp32]%20(TFLite)%20full-inference%20with%20IREE-Vulkan%20@%20SM-G980F%20(GPU-Mali-G77))                                                                                   | 345 (vs. 202, 70.79%↑)    | 345                 | 1                               |
| [DeepLabV3 [fp32] (TFLite) full-inference with IREE-Vulkan @ SM-G980F (GPU-Mali-G77)](https://perf.iree.dev/serie?IREE?DeepLabV3%20[fp32]%20(TFLite)%20full-inference%20with%20IREE-Vulkan%20@%20SM-G980F%20(GPU-Mali-G77))                                                                                   | 130 (vs. 162, 19.75%↓)    | 130                 | 1                               |
| [MobileNetV3Small [fp32,imagenet] (TensorFlow) kernel-execution with IREE-Vulkan @ SM-G980F (GPU-Mali-G77)](https://perf.iree.dev/serie?IREE?MobileNetV3Small%20[fp32,imagenet]%20(TensorFlow)%20kernel-execution%20with%20IREE-Vulkan%20@%20SM-G980F%20(GPU-Mali-G77))                                       | 13 (vs. 15, 13.33%↓)      | 13                  | 0                               |
| [DeepLabV3 [fp32] (TFLite) kernel-execution with IREE-Vulkan @ SM-G980F (GPU-Mali-G77)](https://perf.iree.dev/serie?IREE?DeepLabV3%20[fp32]%20(TFLite)%20kernel-execution%20with%20IREE-Vulkan%20@%20SM-G980F%20(GPU-Mali-G77))                                                                               | 180 (vs. 205, 12.20%↓)    | 209                 | 47                              |
| [MobileNetV2 [fp32,imagenet] (TensorFlow) kernel-execution with IREE-Vulkan @ Pixel-4 (GPU-Adreno-640)](https://perf.iree.dev/serie?IREE?MobileNetV2%20[fp32,imagenet]%20(TensorFlow)%20kernel-execution%20with%20IREE-Vulkan%20@%20Pixel-4%20(GPU-Adreno-640))                                               | 25 (vs. 27, 7.41%↓)       | 25                  | 0                               |
| [PoseNet [fp32] (TFLite) full-inference with IREE-Vulkan @ SM-G980F (GPU-Mali-G77)](https://perf.iree.dev/serie?IREE?PoseNet%20[fp32]%20(TFLite)%20full-inference%20with%20IREE-Vulkan%20@%20SM-G980F%20(GPU-Mali-G77))                                                                                       | 168 (vs. 180, 6.67%↓)     | 168                 | 1                               |
| [MobileNetV2 [fp32,imagenet] (TensorFlow) kernel-execution with IREE-Vulkan @ SM-G980F (GPU-Mali-G77)](https://perf.iree.dev/serie?IREE?MobileNetV2%20[fp32,imagenet]%20(TensorFlow)%20kernel-execution%20with%20IREE-Vulkan%20@%20SM-G980F%20(GPU-Mali-G77))                                                 | 16 (vs. 15, 6.67%↑)       | 16                  | 0                               |
| [MobileNetV2 [fp32,imagenet] (TensorFlow) full-inference with IREE-Vulkan @ SM-G980F (GPU-Mali-G77)](https://perf.iree.dev/serie?IREE?MobileNetV2%20[fp32,imagenet]%20(TensorFlow)%20full-inference%20with%20IREE-Vulkan%20@%20SM-G980F%20(GPU-Mali-G77))                                                     | 76 (vs. 81, 6.17%↓)       | 82                  | 13                              |
| [PoseNet [fp32] (TFLite) kernel-execution with IREE-Vulkan @ SM-G980F (GPU-Mali-G77)](https://perf.iree.dev/serie?IREE?PoseNet%20[fp32]%20(TFLite)%20kernel-execution%20with%20IREE-Vulkan%20@%20SM-G980F%20(GPU-Mali-G77))                                                                                   | 82 (vs. 87, 5.75%↓)       | 82                  | 1                               |
| [MobileNetV3Small [fp32,imagenet] (TensorFlow) full-inference with IREE-Vulkan @ Pixel-4 (GPU-Adreno-640)](https://perf.iree.dev/serie?IREE?MobileNetV3Small%20[fp32,imagenet]%20(TensorFlow)%20full-inference%20with%20IREE-Vulkan%20@%20Pixel-4%20(GPU-Adreno-640))                                         | 17 (vs. 18, 5.56%↓)       | 16                  | 1                               |
| [MobileSSD [fp32] (TFLite) kernel-execution with IREE-Vulkan @ SM-G980F (GPU-Mali-G77)](https://perf.iree.dev/serie?IREE?MobileSSD%20[fp32]%20(TFLite)%20kernel-execution%20with%20IREE-Vulkan%20@%20SM-G980F%20(GPU-Mali-G77))                                                                               | 229 (vs. 238, 3.78%↓)     | 172                 | 75                              |
| [MobileNetV2 [fp32,imagenet] (TensorFlow) full-inference with IREE-Vulkan @ Pixel-4 (GPU-Adreno-640)](https://perf.iree.dev/serie?IREE?MobileNetV2%20[fp32,imagenet]%20(TensorFlow)%20full-inference%20with%20IREE-Vulkan%20@%20Pixel-4%20(GPU-Adreno-640))                                                   | 29 (vs. 30, 3.33%↓)       | 29                  | 0                               |

So basically all improvements, except MobileNetV2 on Mali, which slightly increases by 1ms. I'm fine with that (esp. with MoilbeNetV3 decrease by 2ms, which is also a true improvement).

**Review by MaheshRavishankar — APPROVED** [review]
